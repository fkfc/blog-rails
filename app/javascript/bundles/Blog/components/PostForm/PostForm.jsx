import React from 'react';
import PropTypes from 'prop-types';

export default function PostForm(props) {
  const {
    title,
    content,
    actionPath,
    authenticityToken,
    method,
  } = props;
  const buttonLabel = (method === 'patch') ? 'Update Post' : 'Create Post';

  return (
    <div>
      <form action={actionPath} method="post">
        <input name="_method" readOnly type="hidden" value={method} />
        <input type="hidden" name="authenticity_token" value={authenticityToken} />

        <div className="form-group">
          Title
          <input type="text" id="post[title]" name="post[title]" placeholder="Type the post title here" defaultValue={title} className="form-control" />
        </div>

        <div className="form-group">
          <textarea id="post[content]" name="post[content]" placeholder="Type the post text here" className="form-control" defaultValue={content} rows="10" />
        </div>

        <div className="text-right">
          <input id="submit" name="submit" type="submit" value={buttonLabel} className="btn btn-primary" />
        </div>

      </form>
    </div>
  );
}

PostForm.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  actionPath: PropTypes.string,
  authenticityToken: PropTypes.string.isRequired,
  method: PropTypes.string,
};

PostForm.defaultProps = {
  title: '',
  content: '',
  actionPath: null,
  method: 'post',
};
