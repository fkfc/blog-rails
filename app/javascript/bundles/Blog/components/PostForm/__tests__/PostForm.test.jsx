import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import PostForm from '../PostForm';

describe('Testing the Post Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      authenticityToken: 'token',
    };
  });
  it('Should render correctly', () => {
    const wrapper = shallow(<PostForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
