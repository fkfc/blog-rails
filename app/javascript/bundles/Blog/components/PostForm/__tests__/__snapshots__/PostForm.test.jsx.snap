// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Post Component Should render correctly 1`] = `
<div>
  <form
    action={null}
    method="post"
  >
    <input
      name="_method"
      readOnly={true}
      type="hidden"
      value="post"
    />
    <input
      name="authenticity_token"
      type="hidden"
      value="token"
    />
    <div
      className="form-group"
    >
      Title
      <input
        className="form-control"
        defaultValue=""
        id="post[title]"
        name="post[title]"
        placeholder="Type the post title here"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <textarea
        className="form-control"
        defaultValue=""
        id="post[content]"
        name="post[content]"
        placeholder="Type the post text here"
        rows="10"
      />
    </div>
    <div
      className="text-right"
    >
      <input
        className="btn btn-primary"
        id="submit"
        name="submit"
        type="submit"
        value="Create Post"
      />
    </div>
  </form>
</div>
`;
